ctru-sys/src/lib.rs

```diff
@@ -3,6 +3,7 @@
 #![allow(non_camel_case_types)]
 #![allow(non_snake_case)]
 #![allow(clippy::all)]
+#![allow(unexpected_cfgs)] // Read below why we necessate a check for rust_analyzer.
```

Comment:
I probably would have advocated for adding `rust_analyzer` specifically as a [known cfg](https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html#check-cfg-in-lintsrust-table) something like 
```toml
[lints.rust.unexpected_cfgs]
check-cfg = ['cfg(rust_analyzer)']
```
but not a big difference really.

I guess I'm surprised this isn't a [well-known](https://doc.rust-lang.org/nightly/rustc/check-cfg.html#well-known-names-and-values) value but I guess it's rarer to set than many of those other ones